packages/nuxt/src/pages/module.ts

```diff
@@ -233,7 +233,7 @@ export default defineNuxtModule({
     nuxt.hook('imports:sources', (sources) => {
       const routerImports = sources.find(s => s.from === '#app/composables/router' && s.imports.includes('onBeforeRouteLeave'))
       if (routerImports) {
-        routerImports.from = 'vue-router'
+        routerImports.from = 'vue-router/dist/vue-router.d.ts'
```

Comment:
This is the actual place that these imports should be resolved from, not the declaration file - and will break the usage of these composables.

Instead, the issue is likely with how we are handling `typescript.hoist` elsewhere in the codebase.

Reply:
Sounds good, I'll check how that is being handled!